### ISSUE TITLE
Task 'installDebug' not found in root project ' '.

### ISSUE THREAD

**Issue by CoLagon** [facebook/react-native#13246]
### Description

Hi, I am new to react/react-native and I have a problem that I've been trying to fix for a week. I've created a react native app with the **react-native-cli** with the cmd: `react-native init myApp`. After my app gets created automatically. I run this command: `react-native run-android` and I get this error: 

![runandroidfail](https://cloud.githubusercontent.com/assets/20275733/24572779/aa626d2e-1630-11e7-95e5-520a44e5e732.PNG)

so then to try and fix that I go into the android directory and use the cmd: `gradlew :app:installDebug` and it builds successfully:

![buildsuccessgradle](https://cloud.githubusercontent.com/assets/20275733/24572801/f336ed54-1630-11e7-8c4e-fec3182e6597.PNG)

then I go back into my project directory and run-android again. I expected that this app would run on my android virtual device but I get the same error:

![runandroidagainfail](https://cloud.githubusercontent.com/assets/20275733/24572837/54283726-1631-11e7-9b2b-967c8fc3d0a8.PNG)

So at this point I know there is something seriously wrong so I go back into the android directory and type in this cmd: `gradlew build` and this fails at the bundleReleaseJsAndAssets:

![gradlebuildfail](https://cloud.githubusercontent.com/assets/20275733/24572897/c60731e4-1631-11e7-80a4-2bc9d2e006dc.PNG)

I don't know if this is the problem but something is not working correctly. Like I said I've been trying to fix this for a week and looked at several forums to fix this. I found a issue that has the same build errors as me except this was 2 years ago and he is building the assembleRelease for android with a different version of react: #4124 . 

### Reproduction Steps and Sample Code

If you can get my React-Native app running on your Windows or even your mac let me know what you did. Here is the link to my frustrating app:  https://github.com/CoLagon/React-Native-myApp

### Additional Information

* React Native version: 0.42.3
* Platform: Android
* Development Operating System: Windows 10
* Dev tools: [Android SDK 23.0.1 - 23.0.3]

**Comment by thiensubs** [comment]
What's version of gradle?
EX:
`distributionUrl=https\://services.gradle.org/distributions/gradle-2.14.1-all.zip'`
And what's version of build tool grade?
EX:
 `classpath 'com.android.tools.build:gradle:2.2.3'`

**Comment by CoLagon** [comment]
@thiensubs My classpath is: `classpath 'com.android.tools.build:gradle:2.2.3'`

**Comment by CoLagon** [comment]
@thiensubs and yes my distributionUrl is: `distributionUrl=https\://services.gradle.org/distributions/gradle-2.14.1-all.zip`

**Comment by Air-Miha** [comment]
Hey @CoLagon,

I cloned your project locally, ran `npm install`,  and then `react-native run-android`. I got an error, but not the same one as you. It just said that it couldn't install the app on the device. Then I ran  `./gradlew installDebug` from the `android` directory and got a permission denied error. 

After I did `chmod +x gradlew`, I navigated back to the root directory. I ran `react-native run-android` and it worked. This was on Mac. I'll run your app on my other machine, that has Windows 7, later today. 

I hope this helps!

Did you try running ./gradlew tasks?

**Comment by CoLagon** [comment]
Hey @Air-Miha ,

Thanks for checking out my problem. So I ran cmd gradlew inside the android directory and got this:

![gradlewtaskssuccess](https://cloud.githubusercontent.com/assets/20275733/24582405/27ceaad0-16e4-11e7-95d4-75a716254f49.PNG)

so it seems that `gradlew tasks` builds successfully. Is that weird that `gradlew tasks` builds successfully but `gradlew build` doesn't?

**Comment by Air-Miha** [comment]
@CoLagon, 

I reinstalled everything Android related on my machine with Windows 7. I followed the [official instructions](https://facebook.github.io/react-native/docs/getting-started.html). I was able to run your application on my device. Actually, the installation eventually failed because of insufficient storage, but in your case this would be a good problem to have :). 

Notice that your root project is named **''** in the error message. When I clone it, whatever error message I get, it says root project "myApp". I think your problem is related to your local environment and one of the issues mentioned in this thread:

http://stackoverflow.com/questions/25172006/android-studio-build-fails-with-task-not-found-in-root-project-myproject

Could you try reinstalling everything and going through the official Getting Started documentation? Make sure you don't skip any steps. It will give you a clean start and there's a good chance everything will work.

**Comment by CoLagon** [comment]
@Air-Miha 

the new android sdk update doesn't have a GUI. I downloaded a workaround to getting the GUI back. Did you have a GUI to work off of? or were you doing everything sdk and avd related in a the cmd prompt? I'll delete everything android related I reinstall everything again.

**Comment by CoLagon** [comment]
@Air-Miha 

I Removed everything that was Android related on my Device and reinstalled everything. It still says the same error: I don't know what I'm doing wrong. I have my environment variables set correctly. I have an avd running on my computer. I followed the React documentation and still not luck. It makes no sense that it runs correctly on your computer but on my computer it can't run. I'm so bummed out.

**Comment by CoLagon** [comment]
@Air-Miha 

HEYYYYYYYYYYYYY!!!!!!!! I FIXXED IT!!!!!!!!!!! OMG. I'm so happy. Thank you so much you inspired me to think and fix the problem.  so when you said that it might be my environment, I scratched my head was thinking about my environment variables and things I did to my computer. So about a month ago I went into my registry editor (regedit) and made a new string value in `HKEY_CURRENT_USER\Software\Microsoft\Command Processor` called Autorun so when I open up my command prompt my cmd path would be set to my hard drive (C:). So I removed that and it work! That what was wrong with my computer that entire time. so I guess I learned today to never mess with the regedit. I seriously didn't think that was the problem but it was thank god I removed that.
